jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/EmailDeadlinesBaseTest.java

```diff
@@ -157,6 +157,65 @@ public void testDelayedEmailNotificationOnDeadline() throws Exception {
         assertTrue(list.contains("darth@domain.com"));
     }
     
+    
+    @Test(timeout = 10000)
+    public void testDelayedEmail2NotificationsOnDeadline() throws Exception {  
+        CountDownTaskEventListener countDownListener = new CountDownTaskEventListener(2, false, true);
+        addCountDownListner(countDownListener);
+        
+        Map<String, Object> vars = new HashMap<String, Object>();
+        vars.put("now", new Date());
+
+        Reader reader = new InputStreamReader(getClass().getResourceAsStream(MvelFilePath.DeadlineWith2Notifications));
+        Task task = (Task) TaskFactory.evalTask(reader, vars);
+        
+        taskService.addTask(task, new HashMap<String, Object>());
+        long taskId = task.getId();
+
+        InternalContent content = (InternalContent) TaskModelProvider.getFactory().newContent();
+        
+        Map<String, String> params = fillMarshalSubjectAndBodyParams();
+        ContentData marshalledObject = ContentMarshallerHelper.marshal(task, params, null);
+        content.setContent(marshalledObject.getContent());
+        taskService.addContent(taskId, content);
+        long contentId = content.getId();
+        
+        content = (InternalContent) taskService.getContentById(contentId);
+        Object unmarshallObject = ContentMarshallerHelper.unmarshall(content.getContent(), null);
+        checkContentSubjectAndBody(unmarshallObject);
+
+        // emails should not be set yet
+        assertEquals(0, getWiser().getMessages().size());
+        
+        countDownListener.waitTillCompleted();
+        
+        for (WiserMessage msg : getWiser().getMessages()) {
+            logger.info(msg.getEnvelopeReceiver());
+        }
+        // 1 email with two recipients should now exist
+        assertEquals(2, getWiser().getMessages().size());
+
+        List<String> list = new ArrayList<String>(2);
+        list.add(getWiser().getMessages().get(0).getEnvelopeReceiver());
+        list.add(getWiser().getMessages().get(1).getEnvelopeReceiver());
+
+        assertTrue(list.contains("new-email@domain.com"));
+        assertTrue(list.contains("darth@domain.com"));
+
+        MimeMessage msg = ((WiserMessage) getWiser().getMessages().get(0)).getMimeMessage();
+        assertEquals(myBody, msg.getContent());
+        assertEquals(mySubject, msg.getSubject());
+        assertEquals("from@domain.com", ((InternetAddress) msg.getFrom()[0]).getAddress());
+        assertEquals("replyTo@domain.com", ((InternetAddress) msg.getReplyTo()[0]).getAddress());
+
+        final Address[] recipients = msg.getRecipients(RecipientType.TO);
+        assertNotNull(recipients);
+        assertTrue(recipients.length == 1);
```

Comment:
This is always going to be 1 as the number of the recipients in the email notification is always set to 1 for both deadlines.

Reply:
thats the point